resources/ctia-default.properties

```diff
@@ -166,17 +166,17 @@ ctia.migration.store-keys=malware,tool
 ctia.migration.batch-size=100
 ctia.migration.buffer-size=3
 
-ctia.migration.store.es.default.host=localhost
-ctia.migration.store.es.default.port=9205
-ctia.migration.store.es.default.protocol=http
+#ctia.migration.store.es.default.host=localhost
+#ctia.migration.store.es.default.port=9205
+#ctia.migration.store.es.default.protocol=http
 #ctia.migration.store.es.default.replicas=1
 #ctia.migration.store.es.default.refresh_interval=1s
 #ctia.migration.store.es.default.default_operator=AND
-ctia.migration.store.es.default.shards=1
+#ctia.migration.store.es.default.shards=1
 #ctia.migration.store.es.default.refresh=false
 #ctia.migration.store.es.default.rollover.max_docs=10000000
 #ctia.migration.store.es.default.aliased=true
-ctia.migration.store.es.default.version=5
+#ctia.migration.store.es.default.version=5
```

Comment:
all these properties are redefined in migration tests.
I let them commented to ease local test but avoid interference with production migration properties (for instance in this migration we want to keep source index shard configuration and letting it in default properties would set a new value for the created index).

Reply:
By any chance would it make sense to have separate .properties for production, dev and test?

That way, maybe we wouldn't have to toggle comments in order to try out things.

It's somewhat common to have `dev-resources/`, `test-resources/` folders in a given project.

(If trying this approach, note you need to add said folders to Lein `:resource-paths`)

Reply:
So here is some context. `default.properties` is not a dev file but the default properties that are used in CTIA if not defined in `ctia.properties` that is injected at built time for each envrionement and that could also be overridden by JVM properties and environment variables. [Here](https://github.com/threatgrid/ctia/blob/ac300dfa9a86cb61f49e8a490da0982e7164f234/src/ctia/properties.clj#L218) is the `build-init-config` fn. 

> By any chance would it make sense to have separate .properties for production, dev and test?

We do have separate properties for each CTIA instance (private and public intel) for environment (production, dev and test) and also for each zone for private intel (EU, US, APJC).
Here is the corresponding [tenzin-config PR](https://github.com/threatgrid/tenzin-config/pull/368).

> That way, maybe we wouldn't have to toggle comments in order to try out things.

Indeed, there is no real default properties for a migration (most of them should be specific to each migration), I should simply remove them from `default.properties`. The [migration doc](https://github.com/threatgrid/ctia/blob/master/doc/migration.md#migration-properties) is sufficiently detailed about it.

> It's somewhat common to have `dev-resources/`, `test-resources/` folders in a given project.

@frenchy64 recently rationalized our [properties management](https://github.com/threatgrid/ctia/blob/ac300dfa9a86cb61f49e8a490da0982e7164f234/test/ctia/test_helpers/core.clj#L199) in tests. We also have some [ES fixtures properties](https://github.com/threatgrid/ctia/blob/ac300dfa9a86cb61f49e8a490da0982e7164f234/test/ctia/test_helpers/es.clj#L90) that are also used to override some properties. However we could consider this suggestion for future refactoring about the test properties.

Reply:
Thanks for the explanation! SGTM :)

Maybe I interpreted `I let them commented to ease local tests` incorrectly.

>  I should simply remove them from default.properties. The migration doc is sufficiently detailed about it.

Yes this sounds like a win, since commented-out code can sometimes bring more questions than anyting else. Maybe as a comment inside the properties file, I'd link to https://github.com/threatgrid/ctia/blob/master/doc/migration.md#migration-properties